doc/src/modules/polys/numberfields.rst

```diff
@@ -0,0 +1,289 @@
+
+.. currentmodule:: sympy.polys.numberfields
+
+=============
+Number Fields
+=============
+
+Introduction
+============
+
+  Like many other computations in algebraic number theory, the splitting of
+  rational primes can be treated by *rational* methods only. This fact is very
+  important if computation by automatic computing machinery is considered. Only
+  the knowledge of the irreducible polynomial $f(x)$, a zero of which generates
+  the field in question, is needed.
+
+    Olga Taussky, 1953
+
+
+Concepts like number fields and algebraic numbers are essential to our
+understanding of algebraic number theory, but to the computer the subject is
+all about polynomials: the ring $\mathbb{Q}[x]$ reduced modulo irreducible
+polynomials $f(x) \in \mathbb{Q}[x]$. It thus finds a natural home under the
+:py:mod:`~.polys` module in SymPy.
+
+Various authors (such as Taussky, Zimmer, Pohst and Zassenhaus, or Cohen)
+have articulated the main goals of computational algebraic number theory in
+different ways, but invariably the list centers around a certain essential set
+of tasks. As a goal for the ``numberfields`` module in SymPy, we may set the
+following list, based on [Cohen93]_ (Sec. 4.9.3).
+
+For a number field $K = \mathbb{Q}(\theta)$, whose ring of algebraic integers
+is denoted $\mathbb{Z}_K$, compute:
+
+1. an integral basis of $\mathbb{Z}_K$
+2. the decomposition of rational primes in $\mathbb{Z}_K$
+3. $\mathfrak{p}$-adic valuations for ideals and elements
+4. the Galois group of the Galois closure of $K$
+5. a system of fundamental units of $K$
+6. the regulator $R(K)$
+7. the class number
+8. the structure of the class group $Cl(K)$
+9. decide whether a given ideal is principal, and if so compute a generator.
+
+As a foundation, and to support our basic ability to define and work with
+number fields and algebraic numbers, we also set the following problems:
+
+10. Given an algebraic number -- expressed by radicals and rational operations,
+    or even as a special value of a transcendental function -- determine its
+    minimal polynomial over $\mathbb{Q}$.
+11. Given an irreducible polynomial, distinguish its roots as elements of
+    $\mathbb{C}$ by computing isolating intervals.
+12. The Primitive Element Problem: Given several algebraic numbers
+    $\alpha_1, \ldots, \alpha_m$, compute a single algebraic number $\theta$
+    such that $\mathbb{Q}(\alpha_1, \ldots, \alpha_m) = \mathbb{Q}(\theta)$.
+13. The Field Isomorphism Problem: Decide whether two number fields
+    $\mathbb{Q}(\alpha)$, $\mathbb{Q}(\beta)$ are isomorphic.
+14. The Field Membership Problem: Given two algebraic numbers $\alpha$,
+    $\beta$, decide whether $\alpha \in \mathbb{Q}(\beta)$, and if so write
+    $\alpha = f(\beta)$ for some $f(x) \in \mathbb{Q}[x]$.
+
+At present only a subset of the tasks enumerated in the lists above is yet
+supported in SymPy, and if you are interested in expanding support, you are
+encouraged to contribute!
```

Comment:
This note here is an important one. I think it would be good to make clear that everything needed is in Cohen (if that is the case).

Am I right in thinking that you yourself intend to build on this by implementing some of tasks 4-9?

Reply:
Yes, I would like to implement tasks 4-9, and/or work with others who might be keen on it. And yes, I believe it's all in Cohen. Will add note.